src/java/org/apache/cassandra/db/virtual/CQLMetricsTable.java

```diff
@@ -0,0 +1,87 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.virtual;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import org.apache.cassandra.db.marshal.DoubleType;
+import org.apache.cassandra.db.marshal.UTF8Type;
+import org.apache.cassandra.dht.LocalPartitioner;
+import org.apache.cassandra.metrics.CQLMetrics;
+import org.apache.cassandra.schema.TableMetadata;
+import org.apache.cassandra.cql3.QueryProcessor;
+
+// Virtual table for CQL metrics
+public class CQLMetricsTable extends AbstractVirtualTable
+{
+    public static final String TABLE_NAME = "cql_metrics";
+    public static final String PREPARED_STATEMENTS_COUNT_COL = "prepared_statements_count";
+    public static final String PREPARED_STATEMENTS_EVICTED_COL = "prepared_statements_evicted";
+    public static final String PREPARED_STATEMENTS_EXECUTED_COL = "prepared_statements_executed";
+    public static final String PREPARED_STATEMENTS_RATIO_COL = "prepared_statements_ratio";
+    public static final String REGULAR_STATEMENTS_EXECUTED_COL = "regular_statements_executed";
+    public static final String NAME_COL = "name";
+    public static final String VALUE_ROW = "value";
+
+    private final CQLMetrics cqlMetrics;
+
+    // Default constructor references query processor metrics
+    CQLMetricsTable(String keyspace)
+    {
+        this(keyspace, QueryProcessor.metrics);
+    }
+
+    // For dependency injection
+    @VisibleForTesting
+    CQLMetricsTable(String keyspace, CQLMetrics cqlMetrics)
+    {
+        // create virtual table with this name
+        super(TableMetadata.builder(keyspace, TABLE_NAME)
+                           .kind(TableMetadata.Kind.VIRTUAL)
+                           .partitioner(new LocalPartitioner(UTF8Type.instance))
+                           .addPartitionKeyColumn(NAME_COL, UTF8Type.instance)
+                           .addRegularColumn(PREPARED_STATEMENTS_COUNT_COL, DoubleType.instance)
+                           .addRegularColumn(PREPARED_STATEMENTS_EVICTED_COL, DoubleType.instance)
+                           .addRegularColumn(PREPARED_STATEMENTS_EXECUTED_COL, DoubleType.instance)
+                           .addRegularColumn(PREPARED_STATEMENTS_RATIO_COL, DoubleType.instance)
+                           .addRegularColumn(REGULAR_STATEMENTS_EXECUTED_COL, DoubleType.instance)
+                           .build());
+        this.cqlMetrics = cqlMetrics;
+    }
+
+    @Override
+    public DataSet data()
+    {
+        // populate metrics in the virtual table
+        SimpleDataSet result = new SimpleDataSet(metadata());
+        addRow(result, VALUE_ROW, cqlMetrics);
+
+        return result;
+    }
+
+    private void addRow(SimpleDataSet dataSet, String name, CQLMetrics cqlMetrics)
+    {
+        dataSet.row(name)
+               .column(PREPARED_STATEMENTS_COUNT_COL, Double.valueOf(cqlMetrics.preparedStatementsCount.getValue()))
```

Comment:
Not very sure on the correct/expected format or row/col names for this virtual table, this diff creating below format:
```

cqlsh> select * from system_views.cql_metrics ;

 name  | prepared_statements_count | prepared_statements_evicted | prepared_statements_executed | prepared_statements_ratio | regular_statements_executed
-------+---------------------------+-----------------------------+------------------------------+---------------------------+-----------------------------
 value |                         0 |                           0 |                            0 |                         0 |                          16

(1 rows)

```

Reply:
Agree, me too like to have 5 rows format.
@yifan-c to comment

Reply:
I just looked again in the tables and more specifically in the one that is about to get committed in CASSANDRA-17224 and reminded myself that one was the reason for me to think about the inversion. I think that one makes total sense to 
be in that format and gives a direction for future metrics tables.. So I still feel we need to invert. What do others think? @yifan-c , please, consider the batch_metrics table below:
```
                 name                          | max | p50th | p999th | p99th
-------------------------------+-----+-------+--------+-------
  partitions_per_counter_batch |   0 |     0 |      0 |     0
   partitions_per_logged_batch |   0 |     0 |      0 |     0
 partitions_per_unlogged_batch |   0 |     0 |      0 |     0
```

Reply:
Btw now when the docs migration finished,  we need to consider updating the Virtual Tables page too. Also, we need to add NEWS.txt entry on commit (this one as a note to me and @yifan-c  as probably one of us will commit)

Reply:
Inverted table looks fine to me. @yifan-c to comment.

Reply:
Hi @ekaterinadimitrova2 , good point about the percentiles and thanks for bringing it up. 

I would like to emphasize the importance of **consistency** for the benefit that you all are aware of.

Now, regarding how to place the percentile data, there are existing examples. `CachesTable` is a straightforward one. `TableMetricTables` is another one but more complex. The corresponding columns are created, if we do need to expose the percentile data. Hope it clear things up.

(btw, I am happy to commit this one.)

Reply:
I am confused, do we agree to invert them then? 
I don't see the `CachesTable` as a good comparison to the batch_metrics. I am curious to hear also what @blerer thinks as he was reviewing the second ticket so it is really a good topic for all of us to get aligned with, I totally agree on that.

Reply:
PS @skoppu22 do you mind to update also the docs? We weren't doing it lately due to the docs migration and now we have backlog but there is no reason to postpone doc updates anymore. I can also help with that if there is a need. Just ping me

Reply:
To be clear. I do not think inversion is a good idea. It breaks consistency. And I think having a column for each metrics and rows for metrics sets organizes the data better, especially in the case of having multiple metrics sets. 

I assumed the motivation of inversion was to place the percentile data. So I pointed out that there are existing tables that handles placing percentile data.

Reply:
My view point, inversion of the table makes it to look better. Using metrics as columns makes us to scroll horizontally (columns wrap around to the next line) as they don't fit in screen width well, if you try to see the output I pasted above you need to scroll to see all values. If we just use metrics as rows, we will have only one column, so easy to view and interpret, that's why batch metrics output above feels comfortable to view.

Reply:
Alright. I have presented all my ideas and cannot repeat more. Consistency is the reason I argued to keep the current format. 

Defer to @blerer for the final verdict :D

Reply:
@blerer for the win :D 
My point is I see batch_metrics more consistent with this new one being inverted than if we keep the current format. 
Maybe it is me, let's see what others also think. Otherwise about consistency we are on the same page.

Reply:
This is the PR referred: https://github.com/apache/cassandra/pull/1372
To be consistent, I think its table format needs to be revised.

Reply:
That is what confused me honestly. As we said percentiles in columns but you are against it. Do you suggest all of them in a long header and one row for values?
This is absolutely not needed and user friendly IMHO :( 
I think these two can be done in the proposed format and it will be way more clear and still consistent. 
We can have the format for set of metrics (caches) and the user friendly format for pure one time metric appearance as suggested for batch_metrics and this one. If we follow this rule across new metrics tables, this shouldn't be inconsistent, WDYT?

Reply:
I think the only reason caches was done that way is that we can't make a three-dimensional table

Reply:
Sorry for the confusion. 

So.. I went through the current vtables. As you mentioned, there are 2 formats. The columns are either the metric names or the percentiles. The rows are always the belonging part that has the metrics. For example, 
```sql
# Percentiles as column names. The metric name is the table name.
cqlsh> SELECT * FROM system_views.rows_per_read;

 keyspace_name      | table_name                     | count | max | p50th | p99th
--------------------+--------------------------------+-------+-----+-------+-------
             system |               available_ranges |     1 |   0 |     0 |     0
             system |            available_ranges_v2 |     0 |   0 |     0 |     0
             system |                        batches |    42 |   1 |     1 |     1 

# metric name as column names 
cqlsh> SELECT * FROM system_views.clients;

 address   | port  | connection_stage | driver_name            | driver_version | hostname  | protocol_version | request_count | ssl_cipher_suite | ssl_enabled | ssl_protocol | username
-----------+-------+------------------+------------------------+----------------+-----------+------------------+---------------+------------------+-------------+--------------+-----------
 127.0.0.1 | 63678 |            ready | DataStax Python Driver |         3.25.0 | localhost |                5 |            28 |             null |       False |         null | anonymous
 127.0.0.1 | 63679 |            ready | DataStax Python Driver |         3.25.0 | localhost |                5 |            16 |             null |       False |         null | anonymous
```

To me, the inversion is the third format, where the columns are the percentiles and the rows are the metrics. 
```sql
name     |  value
---------+-------
metric1  |  1
metric2  |  0
```

Maybe I was just being pedantic :| 

What you (both Ekaterina and Shailaja) said make sense to me. I'd +1 to the proposed format. 
Trying to sum up the discussion. For vtable styles, it is preferred to define less columns and have more rows to be better human readable.

Reply:
Happy that we are able to convince you. I have inverted the table format in the code, below is the output from updated code:
```

cqlsh> select * from system_views.cql_metrics ;

 name                         | value
------------------------------+-------
    prepared_statements_count |     0
  prepared_statements_evicted |     0
 prepared_statements_executed |     0
    prepared_statements_ratio |     0
  regular_statements_executed |    17

(5 rows)

```

Reply:
Updated doc as well @yifan-c @ekaterinadimitrova2 please have a look